pyvista/plotting/lights.py

```diff
@@ -434,6 +535,13 @@ def positional(self, state):
             self.hide_actor()
         self.SetPositional(state)
 
+        # add or remove the actor from the renderer
+        if self._renderer is not None:
+            if state:
+                self._renderer.add_actor(self.actor, render=False)
+            else:
+                self._renderer.remove_actor(self.actor, render=False)
+
```

Comment:
I see some issues here:
1. We'd also have to check that we're not reenabling a positional light, leading us to add the same actor multiple times. In the original implementation there was no harm in setting the positional state multiple times. Now we should probably check that the positional state is actually being switched before doing anything.
2. For an actor to render we need a spotlight. So being positional is not enough; if you take a light with 90 degree cone angle and make it positional you'll get the same rendering error. So we also have to factor in the `cone_angle` here, and add the same checks to `cone_angle`'s setter.